common/sgx/verifier.c

```diff
@@ -579,34 +578,33 @@ static oe_result_t _verify_evidence(
 
     if (!memcmp(format_id, &_local_uuid, sizeof(oe_uuid_t)))
     {
-        // evidence_buffer has oe_report_header_t header,
-        // followed by an SGX report for local attestation.
-        oe_report_header_t* report = (oe_report_header_t*)evidence_buffer;
+        // evidence_buffer has an SGX report for local attestation
+        // followed by an optional custom claims buffer.
+        // Note: sgx_report_t has no field that can be checked quickly
+        // to verify it being an SGX report.
 
-        if (evidence_buffer_size < sizeof(*report) ||
-            report->version != OE_REPORT_HEADER_VERSION ||
-            report->report_type != OE_REPORT_TYPE_SGX_LOCAL)
+        if (evidence_buffer_size < sizeof(sgx_report_t))
             OE_RAISE(OE_INVALID_PARAMETER);
 
         format_type = SGX_FORMAT_TYPE_LOCAL;
     }
     else if (!memcmp(format_id, &_ecdsa_uuid, sizeof(oe_uuid_t)))
     {
-        // evidence_buffer has oe_report_header_t header,
-        // followed by an SGX ECDSA-p256 quote.
-        oe_report_header_t* report = (oe_report_header_t*)evidence_buffer;
+        // evidence_buffer has an SGX ECDSA-p256 quote
+        // followed by an optional custom claims buffer.
+        sgx_quote_t* quote = (sgx_quote_t*)evidence_buffer;
 
-        if (evidence_buffer_size < sizeof(*report) ||
-            report->version != OE_REPORT_HEADER_VERSION ||
-            report->report_type != OE_REPORT_TYPE_SGX_REMOTE)
+        if (evidence_buffer_size < sizeof(*quote) + quote->signature_len ||
+            quote->version != SGX_QE3_QUOTE_VERSION ||
+            quote->sign_type != SGX_QL_ALG_ECDSA_P256)
             OE_RAISE(OE_INVALID_PARAMETER);
 
         format_type = SGX_FORMAT_TYPE_REMOTE;
     }
-    else if (!memcmp(format_id, &_ecdsa_report_uuid, sizeof(oe_uuid_t)))
+    else if (!memcmp(format_id, &_legacy_report_uuid, sizeof(oe_uuid_t)))
     {
-        // evidence_buffer has an oe_report_header_t header,
-        // followed by an SGX ECDSA-p256 quote.
+        // evidence_buffer has an oe_report_header_t header
+        // followed by an SGX report or ECDSA-p256 quote.
```

Comment:
What is an `SGX report` in this context? Should it be `SGX quote`?

Reply:
`SGX report` is for local attestation, and `SGX * quote` is for remote attestation.